server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -216,22 +219,22 @@ public DiscoveryNode(
         Version version
     ) {
         if (nodeName != null) {
-            this.nodeName = nodeName.intern();
+            this.nodeName = nodeStringDeduplicator.deduplicate(nodeName);
         } else {
             this.nodeName = "";
         }
-        this.nodeId = nodeId.intern();
-        this.ephemeralId = ephemeralId.intern();
-        this.hostName = hostName.intern();
+        this.nodeId = nodeStringDeduplicator.deduplicate(nodeId);
+        this.ephemeralId = nodeStringDeduplicator.deduplicate(ephemeralId);
+        this.hostName = nodeStringDeduplicator.deduplicate(hostName);
         assert Strings.hasText(hostAddress);
-        this.hostAddress = hostAddress.intern();
+        this.hostAddress = nodeStringDeduplicator.deduplicate(hostAddress);
         this.address = address;
         if (version == null) {
             this.version = Version.CURRENT;
         } else {
             this.version = version;
         }
-        this.attributes = Collections.unmodifiableMap(attributes);
+        this.attributes = Map.copyOf(attributes);
```

Comment:
I am not sure I follow why we want to copy the map here and when deserializing? Wrapping seems less intrusive.

Reply:
Hmm yea this isn't too helpful. I figured this map is often just a single element so this is about as expensive as the wrap call but it doesn't really matter -> reverted :)

Reply:
I think we frequently see around 5-6 entries there so could matter a tiny bit.